File: src/main/java/com/ddd/trainings/proposal/infra/ReviewerRepo.java
package com.ddd.trainings.proposal.infra;

import com.ddd.trainings.proposal.domain.Reviewer;
import com.ddd.trainings.proposal.domain.ReviewerId;
import com.ddd.utils.BaseRepo;
import java.util.List;

public interface ReviewerRepo extends BaseRepo<Reviewer, ReviewerId> {

  List<Reviewer> findByKeywordsOrderByCurrentReviewsCount(String[] value);
}

File: src/Ddd/Trainings/Proposal/Adapters/ReviewerRepo.cs
using System.Collections.Generic;
using System.Linq;
using Ddd.Trainings.Proposal.Domain;
using Ddd.Utils;

namespace Ddd.Trainings.Proposal.Infra
{
    public class ReviewerRepo : BaseRepo<Reviewer, ReviewerId>
    {
        public IEnumerable<Reviewer> FindByKeywordsOrderByCurrentReviewsCount(string[] keywords)
        {
            return _entities.Values
                .Where(r => r.Keywords.Intersect(keywords, StringComparer.OrdinalIgnoreCase).Any())
                .OrderBy(r => r.CurrentReviewsCount);
        }
    }
} 